### PR TITLE
Feature/phpunit 4 main

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | of course, not.

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
Tests went from 7 minutes to 40 (!!!) minutes...

**Comment by trasher** [comment]
> Tests went from 7 minutes to 40 (!!!) minutes...

Yes, this is very long; I have no idea why; but the current goal is to migrate to phpunit; and that's already difficult enough.

**Comment by cedric-anne** [comment]
> > Tests went from 7 minutes to 40 (!!!) minutes...
> 
> Yes, this is very long; I have no idea why; but the current goal is to migrate to phpunit; and that's already difficult enough.

We remove 20 minutes of atoum tests and add 30 minutes of PHPUnit tests. We lose performance, but it is not catastrophic either.

**Comment by AdrienClairembault** [comment]
> We remove 20 minutes of atoum tests and add 30 minutes of PHPUnit tests. We lose performance, but it is not catastrophic either.

I've found where it come from, I'll open another PR so we can win back those 10 minutes.
There is probably more to be found, I've suspected for a while that these tests should not be that slow.

**Comment by cedric-anne** [comment]
> > We remove 20 minutes of atoum tests and add 30 minutes of PHPUnit tests. We lose performance, but it is not catastrophic either.
> 
> I've found where it come from, I'll open another PR so we can win back those 10 minutes. There is probably more to be found, I've suspected for a while that these tests should not be that slow.

You can push directly on this PR.

**Review by cedric-anne — APPROVED** [review]

File: src/CommonDBTM.php [hunk 1]
```diff
@@ -3791,12 +3792,10 @@ public function getNameID($options = [])
...
     final public function searchOptions()
     {
-        static $options = [];
-
         $type = $this->getType();
 
-        if (isset($options[$type])) {
-            return $options[$type];
+        if (isset(static::$search_options_cache[$type])) {
+            return static::$search_options_cache[$type];
```
Comment (cedric-anne) [inline src/CommonDBTM.php:3798]:
```suggestion
        if (isset(self::$search_options_cache[$type])) {
            return self::$search_options_cache[$type];
```

File: src/CommonDBTM.php [hunk 2]
```diff
@@ -3839,6 +3838,7 @@ final public function searchOptions()
             }
         }
 
+        static::$search_options_cache[$type] = $options[$type];
```
Comment (cedric-anne) [inline src/CommonDBTM.php:3841]:
```suggestion
        self::$search_options_cache[$type] = $options[$type];
```

File: src/CommonDBTM.php [hunk 3]
```diff
@@ -6696,4 +6696,9 @@ public static function getSystemSQLCriteria(?string $tablename = null): array
     {
         return [];
     }
+
+    public static function clearSearchOptionCache(): void
+    {
+        static::$search_options_cache = [];
```
Comment (cedric-anne) [inline src/CommonDBTM.php:6702]:
```suggestion
        self::$search_options_cache = [];
```

**Review by AdrienClairembault — APPROVED** [review]

**Pull request merged**